test/v2/fixtures/retrieveBalanceAndTransactions/FinanceDetailsFixture.scala

```diff
@@ -0,0 +1,151 @@
+    documentId = "123456",
+    chargeType = None,
+    mainType = None,
+    taxPeriodFrom = None,
+    taxPeriodTo = None,
+    contractAccountCategory = None,
+    contractAccount = None,
+    documentNumber = None,
+    documentNumberItem = None,
+    chargeReference = None,
+    mainTransaction = None,
+    subTransaction = None,
+    originalAmount = None,
+    outstandingAmount = None,
+    clearedAmount = None,
+    accruedInterest = None,
+    items = Seq[FinancialDetailsItem](financialDetailsItemModel)
+  )
+
+  val mainTypeDownstream: String = "3880"
+
+  val downstreamFinanceDetailFullJson: JsValue = Json.parse(
+    s"""
+       |{
+       |      "taxYear": "${financeDetailsFullObject.taxYear}",
+       |      "documentId": "${financeDetailsFullObject.documentId}",
+       |      "chargeType": "${financeDetailsFullObject.chargeType.get}",
+       |      "mainType": "${mainTypeDownstream}",
+       |      "periodKey": "13RL",
+       |      "periodKeyDescription": "abcde",
+       |      "taxPeriodFrom": "${financeDetailsFullObject.taxPeriodFrom.get}",
+       |      "taxPeriodTo": "${financeDetailsFullObject.taxPeriodTo.get}",
+       |      "businessPartner": "6622334455",
+       |      "contractAccountCategory": "${financeDetailsFullObject.contractAccountCategory.get}",
+       |      "contractAccount": "${financeDetailsFullObject.contractAccount.get}",
+       |      "contractObjectType": "ABCD",
+       |      "contractObject": "00000003000000002757",
+       |      "sapDocumentNumber": "${financeDetailsFullObject.documentNumber.get}",
+       |      "sapDocumentNumberItem": "${financeDetailsFullObject.documentNumberItem.get}",
+       |      "chargeReference": "${financeDetailsFullObject.chargeReference.get}",
+       |      "mainTransaction": "${financeDetailsFullObject.mainTransaction.get}",
+       |      "subTransaction": "${financeDetailsFullObject.subTransaction.get}",
+       |      "originalAmount": ${financeDetailsFullObject.originalAmount.get},
+       |      "outstandingAmount": ${financeDetailsFullObject.outstandingAmount.get},
+       |      "clearedAmount": ${financeDetailsFullObject.clearedAmount.get},
+       |      "accruedInterest": ${financeDetailsFullObject.accruedInterest.get},
+       |      "items":  [$financialDetailsItemDownstreamJson]
+       |}
+       |""".stripMargin
+  )
+
+  val downstreamFinanceDetailMismatchedMainTypeJson: JsValue = Json.parse(s"""
+       |{
+       |      "taxYear": "${financeDetailsFullObject.taxYear}",
+       |      "documentId": "${financeDetailsFullObject.documentId}",
+       |      "mainType": "0000",
+       |      "items":  [$financialDetailsItemDownstreamJson]
+       |}
+       |""".stripMargin)
+
+  val downstreamFinanceDetailMissingMainTypeJson: JsValue = Json.parse(s"""
+       |{
+       |      "taxYear": "${financeDetailsFullObject.taxYear}",
+       |      "documentId": "${financeDetailsFullObject.documentId}",
+       |      "mainType": "0000",
```

Comment:
Suspect this line should be missing (otherwise the value is the same as `downstreamFinanceDetailMismatchedMainTypeJson`). Removing it reveals the issue with the mapping when tests are run (see previous comment).